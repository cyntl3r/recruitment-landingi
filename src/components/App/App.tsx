import { FunctionComponent, lazy, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { fetchArticles } from '../../actions/article';
import { Header } from '../Header/Header';
import './App.styles.css';

const ArticleList = lazy(() =>
  import('../ArticleList/ArticleList').then((module) => ({
    default: module.ArticleList,
  }))
);

const Article = lazy(() =>
  import('../Article/Article').then((module) => ({
    default: module.Article,
  }))
);

const NotFound = lazy(() =>
  import('../NotFound/NotFound').then((module) => ({
    default: module.NotFound,
  }))
);

export const App: FunctionComponent = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchArticles());
  }, []);

  return (
    <div className="app">
      <Header />
      <Switch>
        <Route exact path="/">
          <ArticleList />
        </Route>
        <Route path="/article/:id">
          <Article />
        </Route>
        <Route>
          <NotFound />
        </Route>
      </Switch>
    </div>
  );
};
