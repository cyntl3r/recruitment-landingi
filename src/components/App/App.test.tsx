import React from 'react';
import { render } from '../../testing/test-utils';
import { App } from './App';

test('snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
