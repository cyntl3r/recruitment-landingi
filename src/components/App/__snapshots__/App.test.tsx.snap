// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    class="app"
    style="display: none;"
  >
    <h1
      data-testid="header"
    >
      <a
        href="/"
      >
        Blog
      </a>
       
      
    </h1>
  </div>
</div>
`;
